app/presenters/addl_ids_presenter.rb

```diff
@@ -1,15 +1,23 @@
 # frozen_string_literal: true
 
 class AddlIdsPresenter
-  attr_reader :document, :config
+  attr_reader :fields, :config
 
-  def initialize(document:)
-    @document = document
-    @config = YAML.safe_load(File.open(Rails.root.join('config', 'metadata', 'addl_ids.yml')))
+  def initialize(fields:)
+    @config = YAML.safe_load(File.open(Rails.root.join('config', 'metadata', 'addl_ids.yml'))).symbolize_keys
+    @fields = fields
   end
 
   def terms
-    @config = @config.symbolize_keys
-    @document.slice(*@config.keys)
+    filter(fields: @fields, keys: @config.keys)
+  end
+
+  private
+
+  def filter(fields:, keys:)
```

Comment:
@abelemlih Can you throw this into a module and include it into each of these presenters, since it's used so repetitively?

Reply:
@bwatson78 I added a `FieldPresenter` that is the parent class of all fields presenters, and contains all shared methods.